### ISSUE TITLE
Shortcuts/Macros ideas

### ISSUE THREAD

**Issue by kucera-lukas** [kucera-lukas/miryoku_zmk#6]
1. [x] Complete paired characters (single/double quotes, parens, braces, brackets, bacticks) - PR: #27 
2. [x] ~Move desktop left / right~ (superseded by activities)
3. [x] Email macro - PR: #37 
5. [ ] IDE specific macros (ext doc, rename, usages, ...)
6. [x] Navigate between tabs (browser, terminal, IDE) - PR: #33 
7. [x] Zoom (`+`, `-`), reset - PR: #29, #37 
8. [x] Select word / line - PR: #52 
9. [x] ~Comment lines~ (easy enough with `sym` + `ctrl` + `/`)
10. [x] Join lines (vim `"J"` command) - PR: #52 
11. [x] ~Repeat key~ (currently I don't see a need for it -- not enough keystrokes complex enough to justify repetition)
12. [x] Hold for double/triple symbols - PR: #28 
13. [x] ~`Win + ??` (`D` -- show desktop, `E` -- open explorer, ...)~ (easy enough with homerow mods)
14. [x] ~`CTRL + ??` (`L` -- go to search bar, `RET` -- page break, `Q` -- close window, `U` -- underline, `K` -- cut line, `G` -- go to line, ...)~ (easy enough with homerow mods)
15. [x] Screenshot hold tap - PR: #39 
16. [x] Brightness up down - PR: #37 
17. [x] CTRL, DEL + backspace on hold - PR: #30 
18. [x] Nav improvements on hold - PR: #30 
19. [x] Refresh page (combo switch tab / back-forward?) - PR: #34 
20. [x] Emoji bar - PR: #37 
21. [x] Lock screen - PR: #37 
22. [x] ~Browser back/forward~ (easy enough with alt+left/right)
23. [x] Close tab - PR: #34
24. [x] `CTRL + A` on copy hold - PR: #48 
25. [x] Open/clear/edit clipboard history - PR: #48
26. [x] Mic up/down/mute - PR: #37 
27. [x] Toggle night color - PR: #37 
28. [x] Notifications - PR: #37 
29. [ ] Leader key (https://github.com/urob/zmk-leader-key) - could be useful for [Czech letters](https://github.com/urob/zmk-helpers/blob/main/include/zmk-helpers/unicode-chars/czech.dtsi) (for which I fallback to regular keyboard today) or things like `BOOT` and `SYS`
30. [ ] Adaptive key (https://github.com/urob/zmk-adaptive-key) - seems to complex and I don't see a use-case now
31. [ ] Auto layer (https://github.com/urob/zmk-auto-layer) - improvement over the `continue-list` of `caps_word` and `num_word`
32. [ ] VSCode case transformation - I have a few keybindings gated by `CTRL + K` to tranform casing of selected text in VSCode (https://github.com/kucera-lukas/dotfiles/blob/main/home/dot_config/private_Code/User/keybindings.json). Making these a bit easier would be great as `CTRL + K` is annoying
33. [ ] OS specific differences (Linux, Windows, Mac)

Resources:
- https://getreuer.info/posts/keyboards/macros/index.html
- https://github.com/rafaelromao/keyboards/blob/main/docs/macros.md
- https://github.com/skychil/kombol
- https://raw.githubusercontent.com/seancolsen/qmk_firmware/5fe07c086712f25f2b652aa269264aab8610732c/keyboards/crkbd/keymaps/colsen/diagram.png
- https://github.com/urob/zmk-config/blob/main/config/base.keymap
- https://keymapdb.com/keymaps/alvaro_prieto/
- https://www.jonashietala.se/series/t-34/
